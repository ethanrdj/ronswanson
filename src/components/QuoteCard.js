import React from "react";

import "../styles/QuoteCard.css";
import ron from "../Ron Swanson.svg";

const QuoteCard = ({
  quote,
  setQuote,
  handleRandomQuote,
  searchValue,
  setSearchValue,
  searchQuote,
  handleSearchQuote,
  error,
  setSearchQuote,
}) => {
  const handleChange = (event) => {
    setSearchValue(event.target.value);
  };

  const quoteFunc = () => {
    const quotes = searchQuote.map((quote) => <li>{quote}</li>);
    return quotes;
  };

  return (
    <div className="QuoteCard">
      <h4>Click Ron to get a random quote</h4>
      <img
        className="quote-buttons"
        src={ron}
        alt="ron"
        onClick={() => handleRandomQuote()}
      ></img>
      {quote && (
        <button className="clear-button" onClick={() => setQuote()}>
          X
        </button>
      )}
      <p>{quote}</p>

      <hr></hr>

      <h4>Search for a quote by keyword i.e. bacon</h4>
      <form onSubmit={handleSearchQuote}>
        <input
          type="text"
          className="search-input"
          onChange={handleChange}
          value={searchValue}
        ></input>
        <button>Search</button>
      </form>
      {searchQuote && (
        <button
          onClick={() => {
            setSearchQuote();
            setSearchValue("");
          }}
        >
          clear
        </button>
      )}
      {searchQuote && <ul>{quoteFunc()}</ul>}
      {error && <p>{error}</p>}
    </div>
  );
};

export default QuoteCard;
