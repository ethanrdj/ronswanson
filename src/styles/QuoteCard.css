.QuoteCard {
  color: black;
  padding: 8% 0% 5% 0%;
  border-radius: 5%;
}

.QuoteCard p {
  margin-bottom: 10%;
}

.QuoteCard img {
  height: 10vh;
  margin-right: 25%;
  margin-left: 30%;
}

.search-input {
  border-radius: 5px;
  width: 100%;
  border: none;
  height: 3vh;
  padding: 1%;
  font-size: 1.1em;
}

.QuoteCard button {
  height: 4vh;
  font-weight: bold;
  background: none;
  border: none;
  font-size: 1.1em;
  margin-top: 1vh;
}

.QuoteCard ul {
  list-style-type: none;
  margin-left: -13%;
  width: 100%;
}

.QuoteCard li {
  margin-top: 7%;
  padding-bottom: 7%;
  border-bottom: 2px solid black;
}

.QuoteCard hr {
  width: 75%;
  height: 2px;
  border-width: 0;
  color: black;
  background-color: black;
}
