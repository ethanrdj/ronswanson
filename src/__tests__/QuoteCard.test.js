import React from "react";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import QuoteCard from "../components/QuoteCard";

describe("QuoteCard", () => {
  it("renders the correct component", () => {
    const { asFragment } = render(<QuoteCard quote="This is the quote" />);

    expect(asFragment()).toMatchSnapshot();
  });
});
