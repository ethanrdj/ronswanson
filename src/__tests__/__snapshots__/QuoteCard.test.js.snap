// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuoteCard renders the correct component 1`] = `
<DocumentFragment>
  <div
    class="QuoteCard"
  >
    Random Quote- This is the quote
  </div>
</DocumentFragment>
`;
